modules/openapi-generator/src/main/resources/dart2/serialization/native/native_class.mustache

```diff
@@ -180,37 +225,56 @@ class {{{classname}}} {
     return null;
   }
 
-  static List<{{{classname}}}> listFromJson(dynamic json, {bool emptyIsNull, bool growable,}) =>
-    json is List && json.isNotEmpty
-      ? json.map({{{classname}}}.fromJson).toList(growable: true == growable)
-      : true == emptyIsNull ? null : <{{{classname}}}>[];
+  static List<{{{classname}}}>? listFromJson(dynamic json, {bool growable = false,}) {
+    final result = <{{{classname}}}>[];
+    if (json is List && json.isNotEmpty) {
+      for (final row in json) {
+        final value = {{{classname}}}.fromJson(row);
+        if (value != null) {
+          result.add(value);
+        }
+      }
+    }
+    return result.toList(growable: growable);
+  }
 
   static Map<String, {{{classname}}}> mapFromJson(dynamic json) {
     final map = <String, {{{classname}}}>{};
     if (json is Map && json.isNotEmpty) {
-      json
-        .cast<String, dynamic>()
-        .forEach((key, dynamic value) => map[key] = {{{classname}}}.fromJson(value));
+      json = json.cast<String, dynamic>(); // ignore: parameter_assignments
```

Comment:
can you move this to top of the file? since this gets repeated with every function